doc/hacking.rst

```diff
@@ -443,6 +443,48 @@ Good:
   and are annoying to grep. Further more, they encourage callers not to think
   at all about these optional arguments even if they often should.
 
+- Avoid qualifying modules when accessing fields of records or constructors.
+  Prefer to avoid it altogether if possible or add a type annotation if
+  necessary.
+
+Bad:
+
+.. code:: ocaml
+
+    let result = A.b () in
+    match result.A.field with
+    | B.Constructor -> ...
```

Comment:
There is an OCaml warning for this no? Maybe we should use it. Is it 40?

Reply:
It seems like the warning is enabled by default but dune disables it. Unfortunately, upon trying to figure out why it seems it has been always that way [since jbuilder days](https://github.com/ocaml/dune/commit/cdcd7e907f949fae66a0e57e569c7e0d46f40c3b).

Reply:
I don't see the point in the warning. If the code stop compiling due to a missing type annotation, we just add it. A warning to pre-empt that is just noise.

Reply:
Ah ok I had it backwards nevermind.